Substring error in MessageRecieved handler

There is a specific error that is encountered only a few times, this issue fixes it.

Error:
```
System.ArgumentOutOfRangeException: startIndex cannot be larger than length of string. (Parameter 'startIndex')
   at System.String.ThrowSubstringArgumentOutOfRange(Int32 startIndex, Int32 length)
   at System.String.Substring(Int32 startIndex, Int32 length)
   at Grid.Bot.Events.OnMessage.Invoke(SocketMessage rawMessage) in /_/_work/mfdlabs/grid-bot/34ffcf75b74136b339f592e7c52366157dffe58d/shared/events/Events/OnMessage.cs:line 125
   at Discord.EventExtensions.InvokeAsync[T](AsyncEvent`1 eventHandler, T arg)
   at Discord.EventExtensions.InvokeAsync[T](AsyncEvent`1 eventHandler, T arg)
   at Discord.WebSocket.DiscordSocketClient.TimeoutWrap(String name, Func`1 action)
```

Backtrace:
https://mfdlabs.sp.backtrace.io/p/MFDLABS/triage?_t=W3siZXJyb3IiOnsiZmluZ2VycHJpbnQiOiJhMDVhY2MxOGU5NzA2OTVjYzA4Y2ZiYjI0YzNkYjdiMWQxYzhkZjYyOWYyMGE3M2FkM2NhOTJhZGRkY2Q5MGFmIiwidHgiOjI2ODh9LCJwbHVnaW4iOnsicGx1Z2luSWQiOiJkaXNjb3JkIn0sImluc3RhbmNlIjp7ImFkZHJlc3MiOiJodHRwczovL21mZGxhYnMuc3AuYmFja3RyYWNlLmlvIn19XQ%3D%3D&fingerprint=a05acc18e970695cc08cfbb24c3db7b1d1c8df629f20a73ad3ca92adddcd90af&_tx=a80&time=quarter

New addition to this: It only appears to throw this error IF the bot is directly pinged in the message:

```
[2024-03-03T15:05:43.0261Z][10.124.4.87][grid-bot.nomad.vmminfra.dev][360078081224081409][820111413149433867][360078081224081409][ERROR] Failed to get the command name because: System.ArgumentOutOfRangeException: startIndex cannot be larger than length of string. (Parameter 'startIndex')
   at System.String.ThrowSubstringArgumentOutOfRange(Int32 startIndex, Int32 length)
   at System.String.Substring(Int32 startIndex, Int32 length)
   at Grid.Bot.Events.OnMessage.Invoke(SocketMessage rawMessage) in /_/_work/mfdlabs/grid-bot/22a0ef0100a896d8be29a68951a8c9df2cf9ff3e/shared/events/Events/OnMessage.cs:line 128, Raw Text: <@819791451436482600> hello
```

If you ping another user it will not perform this same action.
This will audit some sort of system to properly parse the prefix in the message.

--- a/shared/events/Events/OnMessage.cs
+++ b/shared/events/Events/OnMessage.cs
@@ -107,6 +107,7 @@ public partial class OnMessage(
 
         _totalMessagesProcessed.Inc();
 
+        // Try catch here temporary until mfdlabs/grid-bot#279 is resolved.
         using var logger = _loggerFactory.CreateLogger(message);
 
         var userIsAdmin = _adminUtility.UserIsAdmin(message.Author);
@@ -115,26 +116,16 @@ public partial class OnMessage(
 
         int argPos = 0;
 
-        if (!message.HasStringPrefix(_commandsSettings.Prefix, ref argPos)
-         && !message.HasMentionPrefix(_client.CurrentUser, ref argPos)) return;
+        if (!message.HasStringPrefix(_commandsSettings.Prefix, ref argPos, StringComparison.OrdinalIgnoreCase)) return;
 
         // Get the name of the command that was used.
         var commandName = message.Content.Split(' ')[0];
         if (string.IsNullOrEmpty(commandName)) return;
 
-        // Try catch here temporary until mfdlabs/grid-bot#279 is resolved.
-        try
-        {
-            commandName = commandName[argPos..];
-            if (string.IsNullOrEmpty(commandName)) return;
-            if (!GetAllowedCommandRegex().IsMatch(commandName)) return;
-            if (!_commandsSettings.PreviousPhaseCommands.Contains(commandName.ToLowerInvariant())) return;
-        }
-        catch (ArgumentException ex)
-        {
-            logger.Error("Failed to get the command name because: {0}, Raw Text: {1}", ex, message.Content);
-            return;
-        }
+        commandName = commandName[argPos..];
+        if (string.IsNullOrEmpty(commandName)) return;
+        if (!GetAllowedCommandRegex().IsMatch(commandName)) return;
+        if (!_commandsSettings.PreviousPhaseCommands.Contains(commandName.ToLowerInvariant())) return;
 
         _totalUsersUsingPreviousPhaseCommands.WithLabels(
             commandName

